pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2274,6 +2288,22 @@ private void updateManagedLedgerConfig() {
         configRegisteredListeners.put(configKey, listener);
     }
 
+    public <T> void registerPublishBufferLimitListener(Consumer<PublishBufferEvent> listener) {
+        publishBufferLimitListeners.add(listener);
+    }
+
+    private void checkPulishBufferLimit() {
+        publishBufferLimitListeners.forEach(listener -> {
+            if (totalPendingBytes.sum() > maxPendingBytes && maxPendingBytes > 0) {
```

Comment:
maxPendingBytes < 0 meaning disable buffer size limit. Please do this check before `sum()` to avoid unnecessary CPU work.